connect/basic-auth-extension/src/main/java/org/apache/kafka/connect/rest/basic/auth/extension/JaasBasicAuthFilter.java

```diff
@@ -174,4 +153,84 @@ public void handle(Callback[] callbacks) throws UnsupportedCallbackException {
                 ));
         }
     }
+
+    private void setSecurityContextForRequest(ContainerRequestContext requestContext, BasicAuthenticationCredential credential) {
+        requestContext.setSecurityContext(new SecurityContext() {
+            @Override
+            public Principal getUserPrincipal() {
+                return () -> credential.getUsername();
+            }
+
+            @Override
+            public boolean isUserInRole(String role) {
+                return false;
+            }
+
+            @Override
+            public boolean isSecure() {
+                return requestContext.getUriInfo().getRequestUri().getScheme().equalsIgnoreCase("https");
```

Comment:
nit: I think getScheme() can return null in the case of relative URIs, in which case we cannot determine from the URI whether the request is secure. I'm not sure if this is a realistic situation or a contrived case.

Looking through the rest of these calls, I think everything else is non-null.

Reply:
Not sure if the URI here can be relative, but if it is not a 100% then optionally a null check can be introduced here:
`return requestContext.getUriInfo().getRequestUri().getScheme() != null && requestContext.getUriInfo().getRequestUri().getScheme().equalsIgnoreCase("https");`

Reply:
I added a null check